recipes/asio-grpc/all/conanfile.py

```diff
@@ -0,0 +1,82 @@
+from conans import ConanFile, CMake, tools
+import os
+
+required_conan_version = ">=1.43.0"
+
+
+class AsioGrpcConan(ConanFile):
+    name = "asio-grpc"
+    description = ("Asynchronous gRPC with Asio/unified executors")
+    homepage = "https://github.com/Tradias/asio-grpc"
+    url = "https://github.com/conan-io/conan-center-index"
+    license = "Apache-2.0"
+    topics = ("cpp", "asynchronous", "grpc", "asio", "asynchronous-programming", "cpp17", "coroutine", "cpp20", "executors")
+    settings = "compiler"
+    no_copy_source = True
+    options = {
+        "backend": ["boost", "asio", "unifex"],
+        "use_boost_container": ["auto", True, False],
+    }
+    default_options = {
+        "backend": "boost",
+        "use_boost_container": "auto",
+    }
+
+    def validate(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, "17")
+
+    def configure(self):
+        if self.options.use_boost_container == "auto":
+            self.options.use_boost_container = self.settings.compiler.get_safe("libcxx") == "libc++" or \
+                (self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) < "9")
+
+    def requirements(self):
+        self.requires("grpc/1.45.2")
+        if self.options.use_boost_container or self.options.backend == "boost":
+            self.requires("boost/1.79.0")
+        if self.options.backend == "asio":
+            self.requires("asio/1.22.1")
+        if self.options.backend == "unifex":
+            self.requires("libunifex/cci.20220430")
+
+    def package_id(self):
+        self.info.header_only()
+        self.info.options.use_boost_container = self.options.use_boost_container
```

Comment:
🤔 i think this can be removed because you already update it in the configure method

```suggestion
```

Reply:
The use_boost_container setting actually changes the source code of asio-grpc. At CMake configure time code is being generated that becomes part of the installation. This is due to a requirement coming from [vcpkg](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md#a-feature-may-replace-polyfills-with-aliases-provided-that-replacement-is-baked-into-the-installed-tree)

Reply:
okay, it affects a header file, right? https://github.com/Tradias/asio-grpc/blob/5d3fd287df3ffde0ff60daf510345a1b38e58410/src/CMakeLists.txt#L27

So it's correct keeping a different package id 👍

Reply:
Thanks for explaining!